core-backend/common/src/lib.rs

```diff
@@ -41,7 +42,26 @@ use gear_core::{
 };
 use gear_core_errors::{ExtError, MemoryError};
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+#[derive(Decode, Encode, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, derive_more::Display)]
+pub struct TrimmedString(String);
+
+// Amount of characters allowed to be thrown as string explanation of the error.
+const TRIMMING_LEN: usize = 1024;
+
+impl<T: Into<String>> From<T> for TrimmedString {
+    fn from(other: T) -> Self {
+        let mut string = other.into();
+
+        if string.len() >= TRIMMING_LEN {
```

Comment:
Not UTF-8 safe I guess as `String::len()` returns bytes length not chars count. A truncated string may become invalid.

Reply:
Fixed it in the most safety, canonical and optimised way, which I found for that case.
Re-review, please